File: android/app/src/main/java/com/mariusdev91/senin/model/WeatherModels.kt
package com.mariusdev91.senin.model

data class WeatherOverview(
    val current: CurrentWeather,
    val hourly: List<HourlyForecast>,
    val daily: List<DailyForecast>,
    val details: WeatherDetails,
    val updatedAtLabel: String,
    val sourceLabel: String,
)

data class CurrentWeather(
    val temperatureC: Int,
    val feelsLikeC: Int,
    val condition: WeatherCondition,
    val summary: String,
    val precipitationChance: Int,
    val humidity: Int,
    val windKph: Int,
    val uvIndex: Int,
)

data class WeatherDetails(
    val airQuality: AirQuality,
    val visibilityKm: Int,
    val pressureHpa: Int,
    val humidity: Int,
    val windKph: Int,
    val windDirectionLabel: String,
    val uvIndex: Int,
    val precipitationChance: Int,
    val sunSchedule: SunSchedule,
)

data class AirQuality(
    val aqi: Int,
    val category: String,
    val primaryPollutant: String,
    val description: String,
)

data class SunSchedule(
    val sunriseLabel: String,
    val sunsetLabel: String,
    val daylightDurationLabel: String,
)

data class SavedLocationPreview(
    val cityId: String,
    val localTimeLabel: String,
    val conditionLabel: String,
    val condition: WeatherCondition,
    val temperatureC: Int,
)

data class HourlyForecast(
    val timeLabel: String,
    val temperatureC: Int,
    val precipitationChance: Int,
    val windKph: Int,
    val condition: WeatherCondition,
)

data class DailyForecast(
    val dayLabel: String,
    val highC: Int,
    val lowC: Int,
    val precipitationChance: Int,
    val condition: WeatherCondition,
)

enum class WeatherCondition {
    Clear,
    PartlyCloudy,
    Cloudy,
    Rain,
    Thunderstorm,
    Snow,
    Mist,
}

File: senin_flutter/lib/src/models/weather_models.dart
enum WeatherCondition {
  clear,
  partlyCloudy,
  cloudy,
  rain,
  thunderstorm,
  snow,
  mist,
}

class WeatherOverview {
  const WeatherOverview({
    required this.current,
    required this.hourly,
    required this.daily,
    required this.updatedAtLabel,
    required this.sourceLabel,
  });

  final CurrentWeather current;
  final List<HourlyForecast> hourly;
  final List<DailyForecast> daily;
  final String updatedAtLabel;
  final String sourceLabel;
}

class CurrentWeather {
  const CurrentWeather({
    required this.temperatureC,
    required this.feelsLikeC,
    required this.condition,
    required this.summary,
    required this.precipitationChance,
    required this.humidity,
    required this.windKph,
    required this.uvIndex,
  });

  final int temperatureC;
  final int feelsLikeC;
  final WeatherCondition condition;
  final String summary;
  final int precipitationChance;
  final int humidity;
  final int windKph;
  final int uvIndex;
}

class HourlyForecast {
  const HourlyForecast({
    required this.timeLabel,
    required this.temperatureC,
    required this.precipitationChance,
    required this.windKph,
    required this.condition,
  });

  final String timeLabel;
  final int temperatureC;
  final int precipitationChance;
  final int windKph;
  final WeatherCondition condition;
}

class DailyForecast {
  const DailyForecast({
    required this.dayLabel,
    required this.highC,
    required this.lowC,
    required this.precipitationChance,
    required this.condition,
  });

  final String dayLabel;
  final int highC;
  final int lowC;
  final int precipitationChance;
  final WeatherCondition condition;
}
